Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -247,6 +247,36 @@ Analytics payload
             return WeaveModuleResult.Success(weaveResults);
         }
 
+        private void RemoveConstructorsBackingFields(TypeDefinition type, HashSet<MetadataToken> backingFields)
+        {
+            // Iterates through all constructor's instructions from the end to start.
+            foreach(var constructor in type.GetConstructors()) {
+                // Index of the most recent strfld <backing_field> instruction
+                var backingFieldInstructionsEnd = -1;
+                for(var i = constructor.Body.Instructions.Count - 1; i >= 0; i--) {
+                    var instruction = constructor.Body.Instructions[i];
+                    
+                    // If it comes across "strfld <backing_field>", 
+                    // it considers this the end index of backing field initializaion instructions.
+                    if (instruction.OpCode == OpCodes.Stfld && instruction.Operand is FieldReference) {
+                        var field = (instruction.Operand as FieldReference);
+                        if(backingFields.Contains(field.MetadataToken)) {
+                            backingFieldInstructionsEnd = i;
+                        }
+                    }
+
+                    // If it comes across "Ldarg 0", 
+                    // it considers this th start index of backing field initializaion instructions.
+                    // And removes all backing field instructions from end to start.
+                    else if (instruction.OpCode == OpCodes.Ldarg_0) {
+                        for(var j = backingFieldInstructionsEnd; j >= i; j--) {
```

Comment:
Setting `backingFieldInstructionsEnd=-1` actually takes care of cases when we never come across any backing fields beforehand, but I could also more explicitly avoid this by doing something like `if (instruction.OpCode == OpCodes.Ldarg_0 && backingFieldInstructionsEnd != -1)`